### PR TITLE
refactor: replace/remove deprecated method calls

### PR DESCRIPTION
**Pull request by rafaelss95**
This:

- Adds `deprecation` rule to the `tslint.json`;
- Replaces all usages of deprecated methods (in majority `tslint` methods);

### CODE REVIEW

File: src/useLifeCycleInterfaceRule.ts [hunk 1]
```diff
@@ -61,28 +61,21 @@ export class ClassMetadataWalker extends Lint.RuleWalker {
     return interfaces;
   }
 
-  private validateMethods(methods: any[], interfaces: string[], className: string) {
+  private validateMethods(methods: ts.ClassElement[], interfaces: string[], className: string) {
     methods.forEach(m => {
-      let n = (<any>m.name).text;
-      if (n && this.isMethodValidHook(m, interfaces)) {
-        let hookName = n.substr(2, n.lenght);
```
Comment (rafaelss95) [inline src/useLifeCycleInterfaceRule.ts]: Fun fact: `lengHt` was mispelled (2 times in this class) :)

Comment (mgechev) [reply src/useLifeCycleInterfaceRule.ts]: Oh!

**Comment by mgechev** [comment]
@rafaelss95 thanks for the cleanup! Would you check why the build is failing?

**Comment by rafaelss95** [comment]
@mgechev so... that was what I asked above, if I change in these 2 places I mentioned, the tests break, I really don't know why.

**Comment by mgechev** [comment]
```
ERROR: /Users/travis/build/mgechev/codelyzer/src/angular/sourceMappingVisitor.ts[107, 18]: createFailure is deprecated: Prefer `addFailureAt` and its variants.
ERROR: /Users/travis/build/mgechev/codelyzer/src/angular/templates/basicTemplateAstVisitor.ts[186, 53]: addFailure is deprecated: Prefer `addFailureAt` and its variants.
```

I guess we should not use deprecated methods.

**Comment by rafaelss95** [comment]
I think you don't understand what I mean. I'm trying to say that if I change these deprecated calls it break some tests. See https://github.com/mgechev/codelyzer/pull/587#issuecomment-385220278

**Comment by rafaelss95** [comment]
@mgechev I've pushed the changes for those leftovers. Now you can see the full log error in travis.

**Comment by rafaelss95** [comment]
bump @mgechev.

**Comment by mgechev** [comment]
@rafaelss95 will take a look in a few minutes.

File: package.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "format:base": "prettier --config ./.prettierrc \"*.{json,md}\" \"src/**/*.{css,scss,ts}\" \"test/**/*.{css,scss,ts}\"",
     "format:check": "npm run format:base -- --list-different",
     "format:fix": "npm run format:base -- --write",
-    "lint": "tslint -c tslint.json \"src/**/*.ts\" \"test/**/*.ts\"",
+    "lint": "tslint -p . -c tslint.json \"src/**/*.ts\" \"test/**/*.ts\"",
```
Comment (mgechev) [inline package.json]: Do we need `-p`? Are we using any rules with type checking?

File: src/angular/sourceMappingVisitor.ts [hunk 3]
```diff
@@ -104,6 +104,7 @@ export class SourceMappingVisitor extends RuleWalker {
 
   createFailure(s: number, l: number, message: string, fix?: Fix): RuleFailure {
     const { start, length } = this.getMappedInterval(s, l);
+    // tslint:disable-next-line:deprecation
     return super.createFailure(start, length, message, fix);
```
Comment (mgechev) [inline src/angular/sourceMappingVisitor.ts]: `createFailure` is deprecated?

File: package.json (hunk 2 above)
Comment (mgechev) [reply package.json]: I see that's required because of the `deprecation` rule?

File: src/angular/sourceMappingVisitor.ts (hunk 3 above)
Comment (rafaelss95) [reply src/angular/sourceMappingVisitor.ts]: Yep, it has been deprecated for over a year. PR https://github.com/palantir/tslint/pull/1952

File: package.json (hunk 2 above)
Comment (rafaelss95) [reply package.json]: Yes, `deprecation` rule requires type information.

File: src/templatesNoNegatedAsyncRule.ts [hunk 4]
```diff
@@ -16,39 +16,41 @@ class TemplateToNgTemplateVisitor extends RecursiveAngularExpressionVisitor {
       return super.visitBinary(expr, context);
     }
 
-    const operator = this.codeWithMap.code.slice(expr.left.span.end, expr.right.span.start);
+    const {
```
Comment (mgechev) [inline src/templatesNoNegatedAsyncRule.ts]: Let's use less of these assignments. A bit hard to follow :-)
